src/js/integrations/backgroundspans.ts

```diff
@@ -0,0 +1,110 @@
+import { Span as SpanClass } from '@sentry/core';
+import type { EventProcessor, Hub, Integration, Transaction } from '@sentry/types';
+import { logger, timestampInSeconds } from '@sentry/utils';
+import type { AppStateStatus } from 'react-native';
+import { AppState } from 'react-native';
+
+export const BACKGROUND_SPAN_OP = 'app.background';
+
+/**
+ * Creates spans for the period of time that the App is in background (not active)
+ *
+ * @experimental
+ */
+export class BackgroundSpans implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'BackgroundSpans';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = BackgroundSpans.id;
+
+  private _currentBackgroundStartTimestamp: number | undefined;
+
+  private _appState: AppState = AppState;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (e: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    if (!this._appState.isAvailable || typeof this._appState.addEventListener !== 'function') {
+      logger.warn('AppState is not available on this platform, BackgroundSpans Integration will not work.');
+      return;
+    }
+
+    this._appState.addEventListener('change', (state: AppStateStatus) => {
+      if (state === <AppStateStatus>'active' && this._currentBackgroundStartTimestamp) {
+        const endTimestamp = timestampInSeconds();
+        const hub = getCurrentHub();
+        const tx = hub.getScope().getTransaction();
+
+        this._createBackgroundSpan(this._currentBackgroundStartTimestamp, endTimestamp, tx);
+        this._currentBackgroundStartTimestamp = undefined;
+      } else if (state === <AppStateStatus>'background' && !this._currentBackgroundStartTimestamp) {
+        this._currentBackgroundStartTimestamp = timestampInSeconds();
+      }
+    });
+  }
+
+  /**
+   * Creates a background span and adds it to the transaction
+   * Patches the transaction's finish method to remove trailing background span
```

Comment:
if it's an idle transaction, can we use https://github.com/getsentry/sentry-javascript/blob/develop/packages/core/src/tracing/idletransaction.ts#L202?

Reply:
Yes, we can check if it is IdleTx and then used the callback.